test/functional/interface_rest.py

```diff
@@ -53,12 +53,13 @@ def skip_test_if_missing_module(self):
 
     def test_rest_request(self, uri, http_method='GET', req_type=ReqType.JSON, body='', status=200, ret_type=RetType.JSON):
         rest_uri = '/rest' + uri
-        if req_type == ReqType.JSON:
-            rest_uri += '.json'
-        elif req_type == ReqType.BIN:
-            rest_uri += '.bin'
-        elif req_type == ReqType.HEX:
-            rest_uri += '.hex'
+        query_string = ''
```

Comment:
Did you consider making `query_string` an optional argument to `test_rest_request()`? It seems a bit contrived to pass the URI without the `req_type` as part of the string, but with the query_string.

Reply:
I agree, that's a more elegant solution. To facilitate dynamic URI creation, I've added a `query_params` kwargs to `test_rest_request()` which is then encoded into a query string.